feat: configurable onboarding config

rarkins opened the pull request:
Adds configuration option `onboardingConfig` that allows the default onboarding config to be extended or replaced via `config.js` or env configuration.

Closes #1521

rarkins commented:
@destroyerofbuilds please let me know if this satisfies your requirement for now

destroyerofbuilds commented on lib/workers/repository/onboarding/branch/rebase.js:
```diff
@@ -1,16 +1,19 @@
+async function rebaseOnboardingBranch(config) {
   logger.debug('Checking if onboarding branch needs rebasing');
+  let contents = await getOnboardingConfig(config);
+  const onboardingBranch = `renovate/configure`;
```
Can this rely on the `branchName` configuration so that if a user of Renovate sets the appropriate environment variable to configure the branch name prefix, it will be used by the on-boarding PR? (This may be out of scope for this change since it's existing behavior)

rarkins replied:
I thought about it and decided it would be too weird to raise an onboarding PR in `renovate/config` and then allow a user to define `branchPrefix` and delete/recreate the onboarding.

However it might be nice for some to allow the branch prefix to be configurable in `config.js` or env when in self-hosting mode, but this is currently not supported and more than just one line.

destroyerofbuilds replied:
> However it might be nice for some to allow the branch prefix to be configurable in config.js or env when in self-hosting mode, but this is currently not supported and more than just one line.

^ That was the specific use case I was thinking about. We don't need that specific feature in the instance of Renovate we run, but perhaps others will want to have the on-boarding PR conform to a naming convention.

destroyerofbuilds commented on lib/workers/repository/onboarding/branch/rebase.js:
```diff
@@ -1,16 +1,19 @@
+    'renovate.json',
+    onboardingBranch
+  );
   const prShort = await platform.getBranchPr(`renovate/configure`);
```
Can this use the `onboardingBranch` variable?

rarkins commented:
By definition it’s a feature for self-hostesses so I’ll wait for one to raise it so I know it’s really needed

destroyerofbuilds commented:
We were able to successfully deploy this enhancement, and all outstanding _Configure Renovate_ pull requests were updated to use our custom configuration.

rarkins commented:
Nice! Be warned, I just fixed a related bug caused by that “update the existing onboarding PRs if possible” functionality though. If you modified the default renovate.json in the PR and then ran Renovate again, it would keep resetting it. Fixed in latest release.
